code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -102,18 +103,26 @@
 			to_chat(src, "<span class='userdanger'>The heat from [I] cauterizes your bleeding!</span>")
 			playsound(src, 'sound/surgery/cautery2.ogg', 70)
 
-		//dismemberment
-		var/dismemberthreshold = (((affecting.max_damage * 2) / max(I.is_sharp(), 0.5)) - (affecting.get_damage() + ((I.w_class - 3) * 10) + ((I.attack_weight - 1) * 15)))
-		if(HAS_TRAIT(src, TRAIT_EASYDISMEMBER))
-			dismemberthreshold -= 50
-		if(I.is_sharp())
-			dismemberthreshold = min(((affecting.max_damage * 2) - affecting.get_damage()), dismemberthreshold) //makes it so limbs wont become immune to being dismembered if the item is sharp
-			if(stat == DEAD)
-				dismemberthreshold = dismemberthreshold / 3
-		if(I.force >= dismemberthreshold && I.force >= 10)
-			if(affecting.dismember(I.damtype))
-				I.add_mob_blood(src)
-				playsound(get_turf(src), I.get_dismember_sound(), 80, 1)
+		var/dismember_limb = FALSE
+		var/weapon_sharpness = I.is_sharp()
```

Comment:
Is this even needed, can we just remove is_sharp and directly read the sharpness variable?

Reply:
We can, but then we are calling the proc to read sharpness multiple times. I have previously been told that I should store any proc calls as a variable so it only has to be called once.

Reply:
It could be read directly but that would require a minor refactor of transforming weapons that isn't super worthwhile